### DISCUSSION TITLE
Quarkus Scheduler: Thread per method?

### DISCUSSION THREAD

**Discussion by justinjohnson02** [quarkusio/quarkus#33125] [Q&A]
Using `quarkus-scheduler` I have 4 methods annotated with `@Scheduled` like the one below.

My question is, how many threads are being used or are allocated for all 4 scheduled methods?

```
@Scheduled(delay = 1, every = "1m", concurrentExecution = SKIP)
void everyMinute() {
     processBlockingJob();
}
```

[a comment by an automated account is not shown]

**Comment by justinjohnson02** [comment]
This is using Quarkus 2.16.6 Final btw

**Comment by mkouba** [comment]
> My question is, how many threads are being used or are allocated for all 4 scheduled methods?

It depends. A thread is never reserved or allocated for a scheduled method. Instead, the default executor for blocking tasks is used. In general, the core thread pool is used, but there are some differences between the production and test/dev modes. Note that [non-blocking scheduled methods](https://quarkus.io/guides/scheduler-reference#non-blocking-methods) are executed on the Vert.x event loop.

**Reply by hatahet** [reply]
Is it possible to inject a custom `ExecutorService`, so that for example we can leverage virtual threads?

**Reply by mkouba** [reply]
Well, you could do anything inside the scheduled method but the question is whether it would be reasonable ;-).

Speaking about the virtual threads - I'm not sure if `@RunOnVirtualThread` is currently supported for scheduled methods but I don't think it is. CC @cescoffier

**Reply by cescoffier** [reply]
It's not supported in @Scheduled method. It's a good idea. Can you open an issue? It's rather simple to implement.

**Reply by cescoffier** [reply]
Note that it will use one virtual thread per method and so may run concurrently (in any order without any guarantee).
